skills/declarative/Calendar/Calendar/dialogs/ListDialog/ListDialog.dialog

```diff
@@ -27,21 +28,24 @@
         {
           "$kind": "Microsoft.EmitEvent",
           "$designer": {
-            "id": "4dyLBN"
+            "id": "4dyLBN",
+            "comment": "Triggers the OnEvent (GetChoices) handler which sets up the various properties needed to set up the list (pageIndex, itemIndex, choices array, etc)."
           },
           "eventName": "GetChoices"
         },
         {
           "$kind": "Microsoft.IfCondition",
           "$designer": {
-            "id": "NbCgE7"
+            "id": "NbCgE7",
+            "comment": "The $doSelection property should be sent in dialog options to indicate whether the list is a choice prompt or a message response. The choice prompt requires the user to choose an option, while the message response shows the list and "
           },
           "condition": "=$doSelection",
           "actions": [
             {
               "$kind": "Microsoft.ChoiceInput",
               "$designer": {
-                "id": "VygDK3"
+                "id": "VygDK3",
+                "comment": "Prompts the user to select from the list of choices. \n\nThe Bot response uses the $isFirstTurn property to show a different message for the first turn of the conversation compared to subsequent turns (for example, after the user navigates to a different page).\n\nThe Property will hold the index of the selected item which will be used to extract the correct item from the choices array.\n\nAllow Interruptions is configured to only allow the following intents with a score > 0.8: Next, Previous, Skip, and Cancel."
```

Comment:
Similar to my comment for $doSelection in GetEventsDialog. In this case, the same comment for $isFirstTurn is used in multiple nodes, a top-level definition would be more concise.